### DISCUSSION TITLE
Different test result between cAT and gordon

### DISCUSSION THREAD

**Discussion by kparysz** [Banno/Gordon#112] [Q&A] [answered]
Hi, I have a Android project where I would like to use gordon. I noticed that gordon fails some tests but connectedAndroidTest passed.

My test case changes date on device before call `launchActivity()` in test. I am using custom test runner where I set IdlingPolicy for RxJava.

Is it possible that gordon has some bug related to that? Is there any method to change date on device trough gordon?

**Comment by joshschriever** [comment, marked as answer]
There is no specific way to change the device date through Gordon.

Gordon [does support using your specified test runner, including a custom one](https://github.com/Banno/Gordon#compatibility-with-android-extension), so I don't think that would be the issue.

Can you try [running each of your test methods individually](https://github.com/Banno/Gordon#filtering) with Gordon? If they each pass individually, then the issue might be related to clearing app state between tests. Gordon does not do anything to clear data between tests (https://github.com/Banno/Gordon/issues/20#issuecomment-576750815, https://github.com/Banno/Gordon/discussions/2#discussioncomment-205645).

**Comment by kparysz** [comment]
@joshschriever Thanks for your reaction. I wasn't aware that gordon is not respecting clearPackage flag from Orchestrator. Indeed, it was a problem with old data on my device. I cleared my storage manually and now everything works perfectlly
